security/pkg/nodeagent/cache/secretcache.go

```diff
@@ -584,21 +584,32 @@ func (sc *SecretCache) rotate(updateRootFlag bool) {
 		// Re-generate secret if it's expired.
 		if sc.shouldRotate(&secret) {
 			atomic.AddUint64(&sc.secretChangedCount, 1)
-
-			// if user not set PROV_CERT Directory Path for client to extract certs from path
-			// we will first check whether the token is expired or not
-			if sc.secOpts.ProvCert == "" && sc.isTokenExpired(&secret) {
+			// Send the notification to close the stream if token is expired, so that client could re-connect with a new token.
+			if isTokenExpired(secret.Token) && !sc.useCertToRotate() {
```

Comment:
I think the logic here should also be:
- if certFetcher != null -> call it
- else - old logic

We won't use cert to rotate or use the token if certFetcher is set.

Reply:
If there is existing valid token, we do not need to call metadata server to get a new one. This can save unnecessary calls to GCE metadata server. Normally token is valid for one hour, but the check here is much more frequent.